packages/components/snackbar/src/SnackbarItem.tsx

```diff
@@ -131,6 +134,19 @@ export const SnackbarItem = forwardRef<HTMLDivElement, PropsWithChildren<Snackba
     const actionBtnFromChildren = findElement('Snackbar.ItemAction')
     const closeBtnFromChildren = findElement('Snackbar.ItemClose')
 
+    useEffect(() => {
+      if (!ref.current) return
+
+      /**
+       * After canceling swipe gesture, we don't want to keep related
+       * data-attributes anymore (to preserve animations & transitions)
+       */
+      if (swipeState === 'cancel' && toast.animation === 'exiting') {
+        ref.current.removeAttribute('data-swipe')
+        ref.current.removeAttribute('data-swipe-direction')
+      }
```

Comment:
Couldn't we use the `onSwipeCancel` callback here, instead of this `useEffect`?

Reply:
Sadly not :disappointed: If I'm not wrong, we don't "listen" to `toast.animation` changes using the `onSwipeCancel` callback.
I tried using `useCallback`, ... with no success. But as said in Slack I'll be glad to get your eyes on this :+1:

Reply:
I checked out your branch, and I don't think you need a `useEffect`, instead you could handle that directly in your render:
```tsx
data-swipe={{ ...(toast.animation === 'exiting' && swipeState) }}
data-swipe-direction={{ ...(toast.animation === 'exiting' && swipeDirection) }}
```

Reply:
After testing it seems that your suggestion doesn't work :sweat_smile: I guess this is because we need to wait the browser to "paint the screen" (CSS) before removing these attributes.

Reply:
Ok copy-pastes were too lazy, but it finally works :+1: Thx @acd02